WordPress/Classes/Utility/Blogging Reminders/BloggingReminders.swift

```diff
@@ -0,0 +1,168 @@
+        case weekdays(_ days: [Weekday])
+    }
+
+    enum Weekday: Int {
+        case sunday = 1 // Keep this at 1 to match Apple's `DateComponents`' weekday number.
+        case monday
+        case tuesday
+        case wednesday
+        case thursday
+        case friday
+        case saturday
+
+        /// The default reminder hour.  In the future we may want to replace this constant with a more customizable approach.
+        ///
+        static let defaultHour = 10
+    }
+
+    /// Singleton for use in the App.
+    ///
+    static let `shared` = BloggingReminders()
+
+    /// The store for persisting our schedule.
+    ///
+    private let store: BloggingRemindersStore
+
+    /// The notification center
+    ///
+    private let notificationCenter: UNUserNotificationCenter
+
+    /// Active schedule.
+    ///
+    var schedule: Schedule {
+        switch store.schedule {
+        case .none:
+            return .none
+        case .weekdays(let days):
+            return .weekdays(days.map({ $0.identifier }))
+        }
+    }
+
+    // MARK: - Initializers
+
+    /// Default initializer.  Allows overriding the blogging reminders store and the notification center for testing purposes.
+    ///
+    init(store: BloggingRemindersStore = .default, notificationCenter: UNUserNotificationCenter = .current()) {
+        self.store = store
+        self.notificationCenter = notificationCenter
+    }
+
+    // MARK: - Scheduling
+
+    /// Main method for scheduling blogging reminder notifications.  This method will take care of scheduling the local notifications and
+    /// persisting the user-defined reminder schedule.
+    ///
+    /// - Parameters:
+    ///     - schedule: the blogging reminders schedule.
+    ///
+    func schedule(_ schedule: Schedule) {
+        unschedule(store.schedule)
+
+        switch schedule {
+        case .none:
+            store.schedule = .none
+        case .weekdays(let days):
+            store.schedule = .weekdays(scheduled(days))
+        }
+    }
+
+    /// Schedules a notifications for the passed days, and returns another array with the days and their
+    /// associated notification IDs.
+    ///
+    /// - Parameters:
+    ///     - weekdays: the weekdays to schedule notifications for.
+    ///
+    /// - Returns: the weekdays with the associated notification IDs.
+    ///
+    private func scheduled(_ weekdays: [Weekday]) -> [BloggingRemindersStore.Weekday] {
+        weekdays.map { scheduled($0) }
+    }
+
+    /// Schedules a notification for the passed day, and returns the day with the associated notification ID.
+    ///
+    /// - Parameters:
+    ///     - weekday: the weekday to schedule a notification for.
+    ///
+    /// - Returns: the weekday with the associated notification ID.
+    ///
+    private func scheduled(_ weekday: Weekday) -> BloggingRemindersStore.Weekday {
+        let notificationID = scheduleNotification(for: weekday)
+        switch weekday {
+        case .monday:
```

Comment:
How about we move this to an initializer on `BloggingRemindersStore.Weekday`? Then you could just do `return BloggingRemindersStore.Weekday(weekday, notificationID)` or similar. It feels like it might fit better there.

Reply:
This is now outdated due to some recent changes, but the solution is not exactly the one proposed here.

Can you check if it looks good now?